src/pages/CodeHandler.js

```diff
@@ -0,0 +1,69 @@
+import React, { useState, useEffect } from "react";
+import { Redirect, useParams } from "react-router";
+import { Flex, Box, Button, Heading, Spinner } from "@chakra-ui/core";
+import { Link as ReactLink } from "react-router-dom";
+
+import ErrorMessage from "../components/ErrorMessage";
+
+import { sendCode } from "../utils/query";
+
+const CodeHandler = () => {
+  const [state, setState] = useState({
+    error: null,
+    isLoading: true,
+  });
+  const { service } = useParams();
+  const getQueryParams = () =>
```

Comment:
Lets move this to `utils/url.js` because this function can be used in other places too.